### ISSUE TITLE
couldscrapper.get is returning js challenge page instead of expected page

### ISSUE THREAD

**Issue by namannehra** [codemanki/cloudscraper#54]
Link to example: https://repl.it/repls/ExpensiveMortifiedDevices
Running it prints the js challenge page instead of expected page

**Comment by n3xtgen** [comment]
I just tried it, and it prints errorType: 1 for me. We have problems with this module too, we use bittrex-market library to connect to Bittrex, an that library uses this one. Last few weeks it almost never connects, only like once out of 10 times it works.

**Comment by namannehra** [comment]
It prints errorType 1 most of the time and other times it prints the challenge page

**Comment by codemanki** [comment]
@namannehra thanks for reporting this. I tried fetching that url 50 times in a row but it still works fine for me and returns an expected html. 
@n3xtgen error type 1 means that cloudflare decided to return a page with recaptcha and unfortunately, this is not something this library can help with

**Comment by n3xtgen** [comment]
Just tried that script few times now again and in 20 tries it printed
10 times errorType: 1
3 times challenge page
7 times actual page

In bittrex-market module cloudscraper is used like this:
```
cloudscraper.get('https://bittrex.com/', (error, response, body) => {
    if(error) {
        throw "Failed to circumvent cloudflare protection: " + error
    }

    this._cloudflareUserAgent = response.request.headers['User-Agent']
    this._cloudflareCookies = response.request.headers['cookie']
    this._cloudflareBypassCallbacksWaiting.forEach(savedCallback => {
        savedCallback()
    })
})
```
I used current version of cloudscraper. It always finishes without error but the `response.request.headers['cookie']` includes only the `__cfduid=d15d3fd866010565a592507ea1c4a26151523666058` cookie and not the other one `cf_clearance=d85d50944325c3d909e42ff4e5ee52dda448dc51-1523668287-10800` which is required for successive request. And the body of cloudscraper result is: 
[bittrex-cf-challenge.txt](https://github.com/codemanki/cloudscraper/files/1909901/bittrex-cf-challenge.txt)
From like 30 tries now, I got that result 25 times, 4 times it returned errorType: 1 and just once it went through...
I even tried to raise "Timeout" in cloudscraper to 15 seconds but it didn't help.

**Comment by n3xtgen** [comment]
Ok now I tried to use a anonymous window in Firefox a few times and it really shows at least 2 successive pages of cloudflare challenge most of the time, sometimes even more, so we need mechanism that will solve whatever number of challenges until returning required page..

**Comment by codemanki** [comment]
This was fixed in 2.0.0. in 3.0.0 `challengesToSolve` will be exposed
